DependencyInjection/Compiler/AnnotationConfigurationPass.php

```diff
@@ -18,32 +18,75 @@
 
 namespace JMS\DiExtraBundle\DependencyInjection\Compiler;
 
-use JMS\DiExtraBundle\Metadata\ClassMetadata;
-use Symfony\Component\DependencyInjection\Alias;
+use JMS\DiExtraBundle\Metadata\MetadataConverter;
+use Metadata\AdvancedMetadataFactoryInterface;
 use JMS\DiExtraBundle\Exception\RuntimeException;
 use JMS\DiExtraBundle\Config\ServiceFilesResource;
 use Symfony\Component\Config\Resource\FileResource;
-use Symfony\Component\DependencyInjection\DefinitionDecorator;
-use Symfony\Component\DependencyInjection\Definition;
 use JMS\DiExtraBundle\Finder\PatternFinder;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 
+/**
+ * Search for annotation usage.
+ */
 class AnnotationConfigurationPass implements CompilerPassInterface
 {
+    /**
+     * @param ContainerBuilder $container
+     */
     public function process(ContainerBuilder $container)
     {
+        /** @var AdvancedMetadataFactoryInterface $factory */
         $factory = $container->get('jms_di_extra.metadata.metadata_factory');
+        /** @var MetadataConverter $converter */
         $converter = $container->get('jms_di_extra.metadata.converter');
         $disableGrep = $container->getParameter('jms_di_extra.disable_grep');
 
         $directories = $this->getScanDirectories($container);
         if (!$directories) {
             $container->getCompiler()->addLogMessage('No directories configured for AnnotationConfigurationPass.');
+
             return;
         }
 
-        $finder = new PatternFinder('JMS\DiExtraBundle\Annotation', '*.php', $disableGrep);
+        /*
+         * process all available patterns
+         */
+        foreach ($this->getPatterns($container) as $pattern) {
+            $this->handlePattern($container, $directories, $pattern, $factory, $converter, $disableGrep);
+        }
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @return \string[]
+     */
+    protected function getPatterns(ContainerBuilder $container)
+    {
+        $patterns = array('JMS\DiExtraBundle\Annotation');
```

Comment:
Would be better to store the container value and then remove it to not have to maintain BC for it.